tidy3d/plugins/design/method.py

```diff
@@ -175,28 +110,466 @@ class MethodGrid(MethodIndependent):
+            save_solutions=True,
+        )
+
+        ga_instance.run()
+
+        if console is not None:
+            solution, solution_fitness, _ = ga_instance.best_solution()
+            console.log(
+                f"Best Result: {solution_fitness}\n"
+                f"Best Parameters: {' '.join([f'{param.name}: {value}' for param, value in zip(parameters, solution)])}\n"
+            )
+
+        # Format output
+        fn_args = [dict(zip(param_keys, val)) for arr in store_parameters for val in arr]
+        results = [val for arr in store_fitness for val in arr]
+
+        return fn_args, results, store_aux, ga_instance
+
+
+class MethodParticleSwarm(MethodOptimize, ABC):
+    """A standard method for performing particle swarm search"""
+
+    n_particles: pd.PositiveInt = pd.Field(
+        ...,
+        title="Number of particles in the swarm.",
+        description="The number of particles to be used in the optimization.",
+    )
+
+    n_iter: pd.PositiveInt = pd.Field(
+        ...,
+        title="Number of iterations.",
+        description="The maxmium number of iterations to run the optimization.",
+    )
+
+    cognitive_coeff: Optional[pd.PositiveFloat] = pd.Field(
+        default=1.5,
+        title="Cognitive coefficient.",
+        description="The cognitive parameter decides how attracted the particle is to its previous best position.",
+    )
+
+    social_coeff: Optional[pd.PositiveFloat] = pd.Field(
+        default=1.5,
+        title="Social coefficient.",
+        description="The social parameter decides how attracted the particle is to the global best position found by the swarm.",
+    )
+
+    weight: Optional[pd.PositiveFloat] = pd.Field(
+        default=0.9,
+        title="Weight.",
+        description="The weight or inertia of particles in the optimization.",
+    )
+
+    ftol: Optional[Union[pd.confloat(ge=0, le=1), Literal[-inf]]] = pd.Field(
+        default=-inf,
+        title="Relative error for convergence.",
+        description="Relative error in `objective_func(best_solution)` acceptable for convergence. See https://pyswarms.readthedocs.io/en/latest/examples/tutorials/tolerance.html for details. Off by default.",
+    )
+
+    ftol_iter: Optional[pd.PositiveInt] = pd.Field(
+        default=1,
+        title="Number of iterations before acceptable convergence.",
+        description="Number of iterations over which the relative error in objective_func is acceptable for convergence.",
+    )
+
+    def run(self, parameters: Tuple[ParameterType, ...], run_fn: Callable, console) -> Tuple[Any]:
+        try:
+            from pyswarms.single.global_best import GlobalBestPSO
+        except ImportError:
+            raise ImportError(
+                "Cannot run particle swarm optimization as pyswarms module not found. Please check installation."
+            )
+
+        # Pyswarms doesn't have a seed set outside of numpy std method
+        np.random.seed(self.rng_seed)
```

Comment:
need to check if `self.rng_seed` is not none?

Reply:
No numpy just ignores a 'None' and picks it's own seed instead.